### PR TITLE
your-editor: 1206 -> 1303

### PR DESCRIPTION
**Pull request by uniquepointer**
###### Motivation for this change
Major version update of yed.

###### Things done
We got MacOS version working, started tagging as promised, changed repo from user to the organization, but aside from that not a whole lot on the nix side of things, a lot on the yed side of things though.
If you want to read yed specific patch notes follow [this link.](https://github.com/your-editor/yed/blob/1303/CHANGELOG.md)

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/your-editor/default.nix [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  version = "1206";
+  version = "1303";
 
   src = fetchFromGitHub {
-    owner = "kammerdienerb";
+    owner = "your-editor";
     repo = "yed";
-    rev = "6cdd99fe1359899b26d8967bd376fd5caa5451eb";
-    sha256 = "0XECSolW/xPXd1v3sv9HbJMWuHGnwCOwmHoPNCUsE+w=";
+    rev = "1b044e5704fcf8de142ae1d5a6eb45728f0d91a2";
```
Comment (figsoda) [inline pkgs/applications/editors/your-editor/default.nix:10]: releases seems to be tagged now
```suggestion
    rev = version;
```

File: pkgs/applications/editors/your-editor/default.nix [hunk 2]
```diff
@@ -22,8 +22,9 @@ stdenv.mkDerivation rec {
     description = "Your-editor (yed) is a small and simple terminal editor core that is meant to be extended through a powerful plugin architecture";
     homepage = "https://your-editor.org/";
     license = with licenses; [ mit ];
-    platforms = platforms.linux;
+    platforms = platforms.unix;
     maintainers = with maintainers; [ uniquepointer ];
     mainProgram = "yed";
   };
 }
+
```
Comment (figsoda) [inline pkgs/applications/editors/your-editor/default.nix:30]:
```suggestion
```

File: pkgs/applications/editors/your-editor/default.nix [hunk 3]
```diff
@@ -1,14 +1,14 @@
-{ lib, stdenv, fetchFromGitHub }:
+{ lib, gccStdenv, fetchFromGitHub }:
```
Comment (figsoda) [inline pkgs/applications/editors/your-editor/default.nix:1]: if this is required, maybe it would be better to override it inside all-packages.nix

**Review by figsoda — CHANGES_REQUESTED** [review]
changelog could be added to meta too

**Comment by uniquepointer** [comment]
Cleaned up everything and squashed accordingly :}

File: pkgs/applications/editors/your-editor/default.nix [hunk 4]
```diff
@@ -21,8 +21,9 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     description = "Your-editor (yed) is a small and simple terminal editor core that is meant to be extended through a powerful plugin architecture";
     homepage = "https://your-editor.org/";
+    changelog = "https://github.com/your-editor/yed/blob/1303/CHANGELOG.md";
```
Comment (figsoda) [inline pkgs/applications/editors/your-editor/default.nix:24]:
```suggestion
    changelog = "https://github.com/your-editor/yed/blob/${version}/CHANGELOG.md";
```

**Comment by figsoda** [comment]
Also can you amend your commit message to `your-editor: 1206 -> 1303`?

**Comment by uniquepointer** [comment]
done!

**Review by figsoda — APPROVED** [review]

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 149675` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>your-editor</li>
  </ul>
</details>

**Comment by uniquepointer** [comment]
thank you!

**Pull request merged**